### PR TITLE
Allow downstream projects to register subcommands

### PR DESCRIPTION
**Pull request by gpaul**
This is a simple PR to allow downstream projects that import dcos-checks to register their own subcommands.

The introduction of the `dep` dependency manager is done in a separate commit to simplify review.

The old set of dependencies appeared to have had several different versions of the `dcos/dcos-go` library pinned, which makes little sense, so I upgraded `dcos/dcos-go` to the latest version.

### CODE REVIEW

File: cmd/checks/executable/executable_check.go [hunk 1]
```diff
@@ -103,7 +103,7 @@ func (c *executableCheck) executableExists(ctx context.Context, cfg *common.CLIC
 		return fmt.Errorf("Choose from valid list of executables %v", keys)
 	}
 
-	if _, _, _, err := exec.Output(ctx, args[0]); err != nil {
+	if _, _, _, err := exec.FullOutput(exec.CommandContext(ctx, args[0])); err != nil {
```
Comment (gpaul) [inline cmd/checks/executable/executable_check.go:106]: This change updates to the latest `dcos/dcos-go` API.

File: cmd/checks/ip/ip_check.go [hunk 2]
```diff
@@ -70,7 +70,7 @@ func (d *detectIPCheck) Run(ctx context.Context, cfg *common.CLIConfigFlags) (st
 		return "", 0, errors.New("path must be set")
 	}
 
-	stdout, stderr, code, err := exec.Output(ctx, d.Path)
+	stdout, stderr, code, err := exec.FullOutput(exec.CommandContext(ctx, d.Path))
```
Comment (gpaul) [inline cmd/checks/ip/ip_check.go:73]: This change updates to the latest `dcos/dcos-go` API.

**Comment by amitaekbote** [comment]
@gpaul the code lgtm. 
If I understand correctly we are now "registering" sub commands, how does this translate to enterprise being able to import all the checks instead of us copying them over? Can you explain the overall context?

**Comment by gpaul** [comment]
@amitaekbote dcos-checks-enterprise can then import the oss dcos-checks repo as a dependency, register its own checks with that repo, then call that repo's Execute. Here's a commit doing exactly that (from a branch I've tested this approach on):

https://github.com/mesosphere/dcos-checks-enterprise/commit/531ebd843a85348179ba11fbece9d2a87a80999b

**Comment by gpaul** [comment]
(please ignore the license in the file headers in that commit ... that branch is still WIP and the license accidentally got copied, too.)

**Review by amitaekbote — APPROVED** [review]

**Comment by amitaekbote** [comment]
Merge on hold till we have the enterprise pr ready

**Review by initcore0 — APPROVED** [review]

**Pull request merged**